gcloud-java-storage/src/test/java/com/google/gcloud/storage/StorageImplTest.java

```diff
@@ -832,6 +859,26 @@ public void testCopyWithOptions() {
   }
 
   @Test
+  public void testCopyWithOptionsFromBlobId() {
+    CopyRequest request = Storage.CopyRequest.builder()
+        .source(BLOB_INFO1.blobId())
+        .sourceOptions(BLOB_SOURCE_GENERATION_FROM_BLOB_ID, BLOB_SOURCE_METAGENERATION)
+        .target(BLOB_INFO1, BLOB_TARGET_GENERATION, BLOB_TARGET_METAGENERATION)
+        .build();
+    StorageRpc.RewriteRequest rpcRequest = new StorageRpc.RewriteRequest(request.source().toPb(),
+        BLOB_SOURCE_OPTIONS_COPY, request.target().toPb(), BLOB_TARGET_OPTIONS_COMPOSE, null);
+    StorageRpc.RewriteResponse rpcResponse = new StorageRpc.RewriteResponse(rpcRequest, null, 42L,
```

Comment:
Nit if `new StorageRpc.RewriteResponse(rpcRequest, null, 42L, false, "token", 21L);` can fit in one line (wrapped) that would be preferred.